### PR TITLE
Update documentation

### PR DESCRIPTION
**Pull request by romanstrobl**
Update of existing documentation and database migration.

### CODE REVIEW

File: docs/Database-Structure.md [hunk 1]
```diff
@@ -40,3 +40,91 @@ Stores user claims.
...
+| `external_id`            | `VARCHAR(255)`                |                                    | Optional external identifier of the stored document (e.g. ID in an external database).                              |
+| `document_data`          | `TEXT`                        | `NOT NULL`                         | Data of the document, encrypted in case encryption is enabled.                                                      |
+| `attributes`             | `TEXT`                        |                                    | Optional map of attributes related to the document, a key-value map serialized into JSON.                           |
+| `encryption_mode`        | `VARCHAR(255)`                | `DEFAULT 'NO_ENCRYPTION' NOT NULL` | Encryption of document data: `NO_ENCRYPTION` means plaintext, `AES_HMAC` for AES encryption with HMAC-based index.  |
+| `timestamp_created`      | `TIMESTAMP WITHOUT TIME ZONE` | `DEFAULT NOW()'`                   | Timestamp of creation of the document.                                                                              |
```
Comment (banterCZ) [inline docs/Database-Structure.md:62]:
```suggestion
| `timestamp_created`      | `TIMESTAMP WITHOUT TIME ZONE` | `DEFAULT NOW()`                    | Timestamp of creation of the document.                                                                              |
```

File: docs/Database-Structure.md [hunk 2]
```diff
@@ -40,3 +40,91 @@ Stores user claims.
...
+| `external_id`            | `VARCHAR(255)`                |                                    | Optional external identifier of the stored document (e.g. ID in an external database).                              |
+| `document_data`          | `TEXT`                        | `NOT NULL`                         | Data of the document, encrypted in case encryption is enabled.                                                      |
+| `attributes`             | `TEXT`                        |                                    | Optional map of attributes related to the document, a key-value map serialized into JSON.                           |
+| `encryption_mode`        | `VARCHAR(255)`                | `DEFAULT 'NO_ENCRYPTION' NOT NULL` | Encryption of document data: `NO_ENCRYPTION` means plaintext, `AES_HMAC` for AES encryption with HMAC-based index.  |
+| `timestamp_created`      | `TIMESTAMP WITHOUT TIME ZONE` | `DEFAULT NOW()'`                   | Timestamp of creation of the record.                                                                                |
```
Comment (banterCZ) [inline docs/Database-Structure.md:86]:
```suggestion
| `timestamp_created`      | `TIMESTAMP WITHOUT TIME ZONE` | `DEFAULT NOW()`                    | Timestamp of creation of the record.                                                                                |
```

File: docs/Database-Structure.md [hunk 3]
```diff
@@ -40,3 +40,91 @@ Stores user claims.
...
+| `external_id`            | `VARCHAR(255)`                |                                    | Optional external identifier of the stored photo (e.g. ID in an external database).                             |
+| `photo_type`             | `VARCHAR(32)`                 | `NOT NULL`                         | Photo type, one of: `person`, `document_front_side`, `document_back_side`, `person_with_document`.              |
+| `photo_data`             | `TEXT`                        | `NOT NULL`                         | Data of the photo, encrypted in case encryption is enabled.                                                     |
+| `encryption_mode`        | `VARCHAR(255)`                | `DEFAULT 'NO_ENCRYPTION' NOT NULL` | Encryption of photo data: `NO_ENCRYPTION` means plaintext, `AES_HMAC` for AES encryption with HMAC-based index. |
+| `timestamp_created`      | `TIMESTAMP WITHOUT TIME ZONE` | `DEFAULT NOW()'`                   | Timestamp of creation of the photo.                                                                             |
```
Comment (banterCZ) [inline docs/Database-Structure.md:106]:
```suggestion
| `timestamp_created`      | `TIMESTAMP WITHOUT TIME ZONE` | `DEFAULT NOW()`                    | Timestamp of creation of the photo.                                                                             |
```

File: docs/Database-Structure.md [hunk 4]
```diff
@@ -40,3 +40,91 @@ Stores user claims.
...
+| `external_id`            | `VARCHAR(255)`                |                                    | Optional external identifier of the stored attachment (e.g. ID in an external database).                             |
+| `attachment_type`        | `VARCHAR(32)`                 | `NOT NULL`                         | Attachment type, one of: `text`, `image_base64`, `binary_base64`.                                                    |
+| `attachment_data`        | `TEXT`                        | `NOT NULL`                         | Data of the attachment, encrypted in case encryption is enabled.                                                     |
+| `encryption_mode`        | `VARCHAR(255)`                | `DEFAULT 'NO_ENCRYPTION' NOT NULL` | Encryption of attachment data: `NO_ENCRYPTION` means plaintext, `AES_HMAC` for AES encryption with HMAC-based index. |
+| `timestamp_created`      | `TIMESTAMP WITHOUT TIME ZONE` | `DEFAULT NOW()'`                   | Timestamp of creation of the attachment.                                                                             |
```
Comment (banterCZ) [inline docs/Database-Structure.md:127]:
```suggestion
| `timestamp_created`      | `TIMESTAMP WITHOUT TIME ZONE` | `DEFAULT NOW()`                    | Timestamp of creation of the attachment.                                                                             |
```

File: docs/Deploying-User-Data-Store.md [hunk 5]
```diff
@@ -31,7 +31,7 @@ There are two roles, that you may assign to the user in the table `uds_authoriti
 ## Deploying User Data Store
 
 Right now, the only standalone war is supported running from console using the `java -jar` command.
-The deployed application is accessible on `http://localhost:8080/user-data-store/`.
+The deployed application is accessible on `http://localhost:8080/user-data-store-server/`.
```
Comment (banterCZ) [inline docs/Deploying-User-Data-Store.md:34]: 🤔 @zcgandcomp Isn't it a problem for the existing customers? I know, this is „only“ documentation, but now I see the consequences of splitting into modules.

File: docs/db/changelog/changesets/user-data-store/1.3.x/20240614-migrate-data-1.3.0.xml [hunk 6]
```diff
@@ -0,0 +1,49 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<databaseChangeLog xmlns="http://www.liquibase.org/xml/ns/dbchangelog"
+                   xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+                   xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-4.9.xsd">
+
+    <changeSet id="1" author="Roman Strobl">
+        <sql>
+            CREATE EXTENSION IF NOT EXISTS "uuid-ossp";
```
Comment (banterCZ) [inline docs/db/changelog/changesets/user-data-store/1.3.x/20240614-migrate-data-1.3.0.xml:8]: Does `CREATE EXTENSION` work for all databases?
```suggestion
    <changeSet id="1" author="Roman Strobl" dbms="postgresql">
        <sql>
            CREATE EXTENSION IF NOT EXISTS "uuid-ossp";
```

**Review by banterCZ — APPROVED** [review]
Looks good. I have several questions to clarify.

File: docs/Deploying-User-Data-Store.md (hunk 5 above)
Comment (zcgandcomp) [reply docs/Deploying-User-Data-Store.md:34]: Yes, it is. 
We will have to be careful with the migration guidelines and prepare for a guided migration of customers under support.

Comment (banterCZ) [reply docs/Deploying-User-Data-Store.md:34]: There is a simple and quick fix by setting the property `server.servlet.context-path`.

Comment (romanstrobl) [reply docs/Deploying-User-Data-Store.md:34]: I will fix it in a separate ticket: https://github.com/wultra/user-data-store/issues/170

**Pull request merged**